### PR TITLE
update ThingType handler

### PR DESCRIPTION
**Pull request by MayankSharma615**
### Description of changes:
Refactoring code and adding better exception handling to conform to all contract test scenarios.

This change includes:

- Changes to all handlers
- update in Unit Tests accordingly

### Testing:
- Unit tests added for coverage
- All contract tests passed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/BaseHandlerStd.java [hunk 1]
```diff
@@ -29,4 +34,17 @@ protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
...
             final ProxyClient<IotClient> proxyClient,
             final Logger logger);
+
+    protected List<Tag> listTags(final ProxyClient<IotClient> proxyClient, final String arn) {
+        String nextToken = null;
+        List<Tag> listOfTags = new ArrayList<>();
+        do {
+            final ListTagsForResourceResponse response = proxyClient.injectCredentialsAndInvokeV2(
+                    Translator.listResourceTagsRequest(arn, nextToken),
+                    proxyClient.client()::listTagsForResource);
```
Comment (aarongup-AWS) [inline aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/BaseHandlerStd.java:44]: Do we need error checking here?

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java [hunk 2]
```diff
@@ -46,63 +54,31 @@
...
-        } else if (e instanceof InternalException) {
-            return HandlerErrorCode.ServiceInternalError;
+            return new CfnInternalFailureException(e);
         } else if (e instanceof ServiceUnavailableException) {
-            return HandlerErrorCode.ServiceInternalError;
+            return new CfnGeneralServiceException(operationName, e);
+        } else if (e instanceof InvalidRequestException || e instanceof InvalidQueryException) {
+            return new CfnInvalidRequestException(e);
+        } else if (e instanceof LimitExceededException) {
+            return new CfnServiceLimitExceededException(ResourceModel.TYPE_NAME, e.getMessage());
```
Comment (aarongup-AWS) [inline aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java:73]: whats the difference between LimitExceededException and ThrottlingException (L76)

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java [hunk 3]
```diff
@@ -206,4 +185,37 @@ static TagResourceRequest tagResourceRequest(final String arn, final Collection<
                 .resourceArn(arn)
                 .tags(tags).build();
     }
+
+    static software.amazon.iot.thingtype.ThingTypeProperties translateThingTypePropertiesToModelObject(
```
Comment (aarongup-AWS) [inline aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java:189]: do we need the full class path here for ThingTypeProperties ?

**Review by aarongup-AWS — COMMENTED** [review]
a couple comments

**Comment by aarongup-AWS** [comment]
can we squash the commits ?

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/BaseHandlerStd.java (hunk 1 above)
Comment (MayankSharma615) [reply aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/BaseHandlerStd.java:44]: Both invocation of this method is inside the try-catch block in - ReadHandler and UpdateHandler. Also, we have the corresponding unit test cases to verify exceptions are translated correctly.

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java (hunk 2 above)
Comment (MayankSharma615) [reply aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java:73]: LimitExceededException is thrown when a service limit - such as number of tags is exceeded. 
ThrottlingException is more relating to rate of requests

https://docs.aws.amazon.com/iot/latest/apireference/API_TagResource.html

File: aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java (hunk 3 above)
Comment (MayankSharma615) [reply aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java:189]: We have 2 definition of ThingTypeProperties is the code base - 
- software.amazon.awssdk.services.iot.model.ThingTypeProperties - defined as part of IoT SDK
- software.amazon.iot.thingtype.ThingTypeProperties - defined as part of CFN resource modeling 

As part of the Translator, we are basically converting the CFN model to IoT SDK model and vice-versa

**Review by aarongup-AWS — APPROVED** [review]

**Review by DerekwanAmazon — APPROVED** [review]

**Pull request merged**